### PR TITLE
Migrate setup.py to pyproject.toml

### PR DESCRIPTION
**Pull request by Kota-Karthik**
Hello @rootCircle 
I had changed the setup.py to pyproject.toml as per you mentioned in the issue #3 
I am attaching screenshot of the successful working of the "codemark" command in the terminal for your reference:-
image

You can also check that I am running this command after the deletion of setup.py
I am attaching the screenshot for your reference:-
image

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+[tool.poetry]
+name = "codemark"
+version = "0.2.1"
+description = "CodeMark CLI is a command-line interface that helps you manage your coding assignments and tests.\\nWith this tool, you can easily initialize the configuration, list assignments, fetch and check your code, submit your code for grading, and get AI-powered error recommendations."
+authors = ["rootCircle"]
+license = "Apache 2.0"
```
Comment (rootCircle) [inline pyproject.toml]: Consider using SPDX names in license fields.
https://spdx.org/licenses/Apache-2.0.html

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+[tool.poetry]
+name = "codemark"
+version = "0.2.1"
+description = "CodeMark CLI is a command-line interface that helps you manage your coding assignments and tests.\\nWith this tool, you can easily initialize the configuration, list assignments, fetch and check your code, submit your code for grading, and get AI-powered error recommendations."
+authors = ["rootCircle"]
```
Comment (rootCircle) [inline pyproject.toml]: For author, you can prefer using 
```toml
authors = ["Lab Rat <dev.frolics@gmail.com>"]
```

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+license = "Apache 2.0"
+readme = "README.md"
+
```
Comment (rootCircle) [inline pyproject.toml:17]: Missing some important meta like github url, maintainers, keywords and other package_data as in setup.py
>    package_data={'codemark': ['firebase/res/service-account-file.json']}


https://packaging.python.org/en/latest/specifications/declaring-project-metadata/

**Comment by rootCircle** [comment]
Can you update README.md for relevant changes too please?

**Review by rootCircle — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 3 above)
Comment (Kota-Karthik) [reply pyproject.toml:17]: @rootCircle 
Package data has been added to the TOML file

Comment (rootCircle) [reply pyproject.toml:17]: I believe other meta content like urls, and maintainers and keywords still need to added in! :-)
follow these for help : https://packaging.python.org/en/latest/specifications/declaring-project-metadata/

**Review by rootCircle — APPROVED** [review]

**Comment by rootCircle** [comment]
LGTM :+1: Merged
Congrats on your PR! :tada:

**Pull request merged**